### PR TITLE
CounterControl bug fix

### PR DESCRIPTION
**Pull request by Carlm-MS**
fix for NullReferenceExceptions and missing dependencies

### CODE REVIEW

File: src/OrleansCounterControl/OrleansCounterControl.csproj [hunk 1]
```diff
@@ -45,8 +45,24 @@
     <WarningsAsErrors>4014</WarningsAsErrors>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="Microsoft.CodeAnalysis, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (jthelin) [inline src/OrleansCounterControl/OrleansCounterControl.csproj]: These should be "soft" assembly name references - see "External Assembly References"
http://dotnet.github.io/orleans/Contributing

Comment (sergeybykov) [inline src/OrleansCounterControl/OrleansCounterControl.csproj]: @jthelin I thought we concluded that there is no reason to fight the tools and manually remove version data from references in .csproj files.

Comment (jthelin) [inline src/OrleansCounterControl/OrleansCounterControl.csproj]: You keep asserting that "conclusion" @sergeybykov But as far as i can recall [or find using search] there has never been anything documented and/or agreed decision that changes & reverses the project development practices that are written down on this project's "Contributing" pages:
http://dotnet.github.io/orleans/Contributing

The thread below is the only canonical discussion i could find about assembly soft references. 
That thread even shows specific concrete examples where the VS / NuGet tools was demonstrably doing **wrong** things -- and that conclusion was not just mine but confirmed / agreed by others too. 
https://github.com/dotnet/orleans/pull/1115

Do you have a citation for the asserted "conclusion" ?

Comment (sergeybykov) [inline src/OrleansCounterControl/OrleansCounterControl.csproj]: I guess I considered this comment by @jdom a "conclusion" - https://github.com/dotnet/orleans/pull/1115#discussion_r47028940. I don't recall any example of a real issues with the way NuGet puts references to .csproj files resented after that.

Comment (veikkoeeva) [inline src/OrleansCounterControl/OrleansCounterControl.csproj]: It looks to me people arguing for using what Nuget is doing by default and avoid both extra work and go along with the tool. That being written, if I read this right, there should be a decision whether we should have soft references or not before moving to the new system and after that it doesn't matter. A point weighing in decision is if it matters in the meantime. Or did I stir up a hornet's nest now? :)

Comment (sergeybykov) [inline src/OrleansCounterControl/OrleansCounterControl.csproj]: I think we need to remove the "External Assembly References" section from http://dotnet.github.io/orleans/Contributing.

**Comment by sergeybykov** [comment]
Thanks you, @Carlm-MS!

File: src/OrleansCounterControl/OrleansCounterControl.csproj (hunk 1 above)
Comment (jthelin) [inline src/OrleansCounterControl/OrleansCounterControl.csproj]: Thanks for locating that comment link @sergeybykov

I agree that the project working practices should accurately reflect the real "rules" that are expected to be applied. I won't push back if you want to make those changes to the doc.
http://dotnet.github.io/orleans/Contributing

Comment (sergeybykov) [inline src/OrleansCounterControl/OrleansCounterControl.csproj]: Thanks @jthelin. I updated http://dotnet.github.io/orleans/Contributing.

**Pull request merged**